filebeat/inputsource/udp/server.go

```diff
@@ -0,0 +1,79 @@
+package udp
+
+import (
+	"net"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common/atomic"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Config options for the UDPServer
+type Config struct {
+	Host           string        `config:"host"`
+	MaxMessageSize int           `config:"max_message_size"`
+	Timeout        time.Duration `config:"timeout"`
+}
+
+// Server creates a simple UDP Server and listen to a specific host:port and will send any
+// event received to the callback method.
+type Server struct {
+	running  atomic.Bool
+	config   *Config
+	callback func(data []byte)
+	listener net.PacketConn
+}
+
+// New returns a new UDPServer instance.
+func New(config *Config, callback func(data []byte)) *Server {
+	return &Server{
+		running:  atomic.MakeBool(false),
+		config:   config,
+		callback: callback,
```

Comment:
You could create a `Logger` instance for the server to use.

```
  log: logp.NewLogger("udp").With("address", config.Host),
```

Then anywhere within this object that you do logging you would use `s.log.Xyz()`. This gives you a named logger with context plus gives you the ability to use the structured logging methods if needed.